### PR TITLE
antigen: init at 2.2.1

### PR DESCRIPTION
**Pull request by bugworm**
###### Motivation for this change
Want to add antigen. Also I've planned to add module for it, similar to oh-my-zsh

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by 0xABAB — APPROVED** [review]

**Comment by bugworm** [comment]
I don't sure about that kind of dowloading script, because it will cause old versions be the same as last version, but it's recommended way to install it

File: pkgs/shells/antigen/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  version = "2.2.1";
+  name = "antigen";
+
+  src = fetchurl {
+    url = "git.io/antigen";
```
Comment (Mic92) [inline pkgs/shells/antigen/default.nix]: This source is not stable. Is there a different source, which does not break with every upstream update?

Comment (bugworm) [inline pkgs/shells/antigen/default.nix]: There are releases, I can fetch file from there. Or clone a repo and copy all needed stuff from there

Comment (bugworm) [inline pkgs/shells/antigen/default.nix]: Is fetching from `https://github.com/zsh-users/antigen/releases/download/v${version}/antigen.zsh` will be ok?

Comment (Mic92) [inline pkgs/shells/antigen/default.nix]: Either from the release page, or using `fetchFromGitHub`.

**Comment by Mic92** [comment]
Thanks. I did not notice the update.

**Pull request merged**